cmd/podman/system/connection/add.go

```diff
@@ -202,30 +204,71 @@ func getUserInfo(uri *url.URL) (*url.Userinfo, error) {
 	return url.User(usr.Username), nil
 }
 
-func getUDS(cmd *cobra.Command, uri *url.URL) (string, error) {
-	var signers []ssh.Signer
+func getUDS(cmd *cobra.Command, uri *url.URL, iden string) (string, error) {
+	cfg, err := Validate(uri, iden)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to validate")
+	}
+	dial, err := ssh.Dial("tcp", uri.Host, cfg)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to connect")
+	}
+	defer dial.Close()
 
+	session, err := dial.NewSession()
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to create new ssh session on %q", uri.Host)
+	}
+	defer session.Close()
+
+	// Override podman binary for testing etc
+	podman := "podman"
+	if v, found := os.LookupEnv("PODMAN_BINARY"); found {
+		podman = v
+	}
+	run := podman + " info --format=json"
+
+	var buffer bytes.Buffer
+	session.Stdout = &buffer
+	if err := session.Run(run); err != nil {
+		return "", err
+	}
+
+	var info define.Info
+	if err := json.Unmarshal(buffer.Bytes(), &info); err != nil {
+		return "", errors.Wrapf(err, "failed to parse 'podman info' results")
+	}
+
+	if info.Host.RemoteSocket == nil || len(info.Host.RemoteSocket.Path) == 0 {
+		return "", errors.Errorf("remote podman %q failed to report its UDS socket", uri.Host)
+	}
+	return info.Host.RemoteSocket.Path, nil
+}
+
+// Validate will take a ssh url and an identity key (rsa and the like) and ensure the information you gave is valid
+// iden and url can also come from image scp, iden in this cae might be blank if using local or remote client or if no identification is required
```

Comment:
> `// Validate will take a ssh url and an identity key (rsa and the like) and ensure the information you gave is valid`

That’s not quite what the function does; it seems to me; it primarily returns a `ssh.ClientConfig`; in the process it validates `iden` but not really `uri`. So it should probably be both differently-documented and renamed.

---

Go conventions are to use present tense for documentation. The "you" might be better avoided, (especially if the code is successful and widely used,) many readers will have nothing to do with the caller that provided the information; `the information is valid` works just as well.

---

```suggestion
// iden and url can also come from image scp, iden in this case might be blank if using local or remote client or if no identification is required
```

but `iden` can be empty also with the `connection add` command, AFAICS, and if using a local client in scp, why would this be called at all?

Reply:
some ssh connections might not requires an identity key if sshd is configured in a certain way so iden could potentially be blank coming from scp I believe. I renamed the function to `ValidateAndConfigure` and edited the comment

Reply:
I think what I’m ultimately saying about the remote/local/scp notes is that this function is shared, and doesn’t care about whether it is called from image scp (and that’s good BTW). “`iden` can be blank to mean no identity key” is sufficient, and universal for all callers; no need to discuss scp.